x-pack/plugins/observability_ai_assistant/public/components/insight/insight.tsx

```diff
@@ -122,14 +134,75 @@ function ChatContent({
   );
 }
 
+function PromptEdit({
+  initialPrompt,
+  onSend,
+  onCancel,
+}: {
+  initialPrompt: string;
+  onSend: (updatedPrompt: string) => void;
+  onCancel: () => void;
+}) {
+  const [prompt, setPrompt] = useState(initialPrompt);
+
+  return (
+    <EuiFlexGroup alignItems={'center'}>
+      <EuiFlexItem grow={true}>
+        <EuiTextArea
+          data-test-subj="observabilityAiAssistantInsightEditPromptTextArea"
+          inputRef={(textarea) => {
+            if (textarea) {
+              setTimeout(() => textarea.focus());
+            }
+          }}
+          fullWidth={true}
+          defaultValue={prompt}
+          onChange={(ev) => {
+            setPrompt(ev.target.value);
+          }}
+        />
+      </EuiFlexItem>
+      <EuiFlexItem grow={false}>
+        <EuiButtonIcon
+          aria-label={i18n.translate('xpack.observabilityAiAssistant.insight.cancelPromptEdit', {
+            defaultMessage: 'Cancel',
+          })}
+          data-test-subj="observabilityAiAssistantInsightCancelEditPromptButtonIcon"
+          iconType="cross"
+          display="base"
+          color="danger"
+          size="m"
+          onClick={onCancel}
+        />
+      </EuiFlexItem>
+      <EuiFlexItem grow={false}>
+        <EuiButtonIcon
+          aria-label={i18n.translate('xpack.observabilityAiAssistant.insight.sendPromptEdit', {
+            defaultMessage: 'Send prompt',
+          })}
+          data-test-subj="observabilityAiAssistantInsightSendEditPromptButtonIcon"
+          iconType="kqlFunction"
+          display="fill"
+          size="m"
+          onClick={() => onSend(prompt)}
+        />
+      </EuiFlexItem>
+    </EuiFlexGroup>
+  );
+}
+
 export interface InsightProps {
   messages: Message[];
   title: string;
   dataTestSubj?: string;
 }
 
 export function Insight({ messages, title, dataTestSubj }: InsightProps) {
+  const [initialMessages, setInitialMessages] = useState(messages);
+  const [isEditingPrompt, setEditingPrompt] = useState(false);
+  const [isInsightOpen, setInsightOpen] = useState(false);
   const [hasOpened, setHasOpened] = useState(false);
+  const [updatedPrompt, setUpdatedPrompt] = useState(false);
```

Comment:
nit
```suggestion
  const [didUpdatePrompt, setDidUpdatePrompt] = useState(false);
```

Reply:
fixed in https://github.com/elastic/kibana/pull/176086/commits/5f8cd0724c253393609bb3091d23c011b6a2d288